package NYT.NChunkManager.NProto;

import "chunk_manager_rpc.proto";

////////////////////////////////////////////////////////////////////////////////

message TMsgUnregisterHolder
{
    required int32 HolderId = 1;
}

message TMsgAddChunk
{
    required bytes ChunkId = 1;
    required bytes TransactionId = 2;
}

message TMsgRemoveChunk
{
    required bytes ChunkId = 1;
}

message TMsgHeartbeatRequest
{
    required int32 HolderId = 1;
    required THolderStatistics Statistics = 2;
    repeated TChunkInfo AddedChunks = 3;
    repeated bytes RemovedChunks = 4;
}

message TMsgHeartbeatResponse
{
    required int32 HolderId = 1;
    repeated TJobStartInfo StartedJobs = 2;
    repeated bytes StoppedJobs = 3;
}

////////////////////////////////////////////////////////////////////////////////

